docs/docs/grpc.md

```diff
@@ -18,12 +18,134 @@ main = runGRpcApp 8080 "helloworld" quickstartServer
 
 ## Building a client
 
--- TODO:
+Right now there are two options for building clients: using records or with `XTypeApplications`. To give a proper introduction to both options let's consider in detail an example client for the following services:
+
+```protobuf
+service Service {
+  rpc getPerson (PersonRequest) returns (Person);
+  rpc newPerson (Person) returns (PersonRequest);
+  rpc allPeople (google.protobuf.Empty) returns (stream Person);
+}
+```
+
+Regardless of the approach we decide to use, we can construct a basic CLI for the client this way:
+
+```haskell
+import System.Environment
+
+main :: IO ()
+main = do
+  let config = grpcClientConfigSimple "127.0.0.1" 8080 False
+  Right client <- setupGrpcClient' config
+  args <- getArgs
+  case args of
+    ["watch"]       -> watching client
+    ["get", idp]    -> get client idp
+    ["add", nm, ag] -> add client nm ag
+    _               -> putStrLn "unknown command"
+```
+
+Where `watch`, `get` and `add` are the only valid 3 commands that our CLI is going to accept and call each respective service.
 
 ### Using records
 
--- TODO:
+This option is a bit more verbose but it's also more explicit with the types and _"less magic"_ than the one with `TypeApplications`.
+
+We need to define a new record type (hence the name) that declares the services our client is going to consume:
+
+```haskell
+import GHC.Generics (Generic)
+import Mu.GRpc.Client.Record
+
+data Call = Call
+  { getPerson :: MPersonRequest -> IO (GRpcReply MPerson)
+  , newPerson :: MPerson -> IO (GRpcReply MPersonRequest)
+  , allPeople :: IO (ConduitT () (GRpcReply MPerson) IO ())
+  } deriving Generic
+```
+
+Note that we had to derive `Generic`. We also need to tweak a little bit our `main` function:
+
+```diff
+main :: IO ()
+main = do
+   let config = grpcClientConfigSimple "127.0.0.1" 1234 False
+-  Right client <- setupGrpcClient' config
++  Right grpcClient <- setupGrpcClient' config
++  let client = buildService @Service @"" grpcClient
```

Comment:
I'm a bit confused by `@""`. `""` is a singleton type?

Is there some other documentation where I can check the signature of `buildService`? If so, a link to that would be good.

Reply:
I agree, that bit is quite confusing, [here's the type signature](https://github.com/higherkindness/mu-haskell/blob/master/grpc/client/src/Mu/GRpc/Client/Record.hs#L46-L50) of `buildService` but you are not gonna like it 😂. I'll wait for Alejandro's feedback when he comes back to see what else to add/clarify on the docs 😉

Reply:
Art.

Reply:
The `@""` is a type-level string which defines the prefix of the service, something akin to the route where the service is located. I have been doubting for a while whether a type-level string is the best choice, or we should use a normal string (well, actually a `ByteString`) here instead.

We still have time to change the API, and we can even provide both versions.